src/placement/tests/place_obj_common.h

```diff
@@ -128,4 +143,72 @@ extend_test_pool_map(struct pool_map *map, uint32_t nnodes, d_rank_list_t *rank_
 
 bool
 is_max_class_obj(daos_oclass_id_t cid);
+
+static inline void
+verbose_msg(char *msg, ...)
+{
+	if (g_verbose) {
+		va_list vargs;
+
+		va_start(vargs, msg);
+		vprint_message(msg, vargs);
+		va_end(vargs);
+	}
+}
+
+static inline void
+gen_maps(int num_pds, int fdoms_per_pd, int nodes_per_domain, int vos_per_target,
+	 struct pool_map **po_map, struct pl_map **pl_map)
+{
+	*po_map = NULL;
+	*pl_map = NULL;
+	gen_pool_and_placement_map(num_pds, fdoms_per_pd, nodes_per_domain, vos_per_target,
+				   PL_TYPE_JUMP_MAP, po_map, pl_map);
+	assert_non_null(*po_map);
+	assert_non_null(*pl_map);
+}
+
+static inline void
+gen_oid(daos_obj_id_t *oid, uint64_t lo, uint64_t hi, daos_oclass_id_t cid)
+{
+	int rc;
+
+	oid->lo = lo;
+	/* make sure top 32 bits are unset (DAOS only) */
+	oid->hi = hi & 0xFFFFFFFF;
+	rc = daos_obj_set_oid_by_class(oid, 0, cid, 0);
+	assert_rc_equal(rc, cid == OC_UNKNOWN ? -DER_INVAL : 0);
+}
+
+#define assert_placement_success_print(pl_map, cid, pda)			\
+	do {									\
+		daos_obj_id_t __oid;						\
+		struct pl_obj_layout *__layout = NULL;				\
+		gen_oid(&__oid, 1, UINT64_MAX, cid);				\
+		assert_success(plt_obj_place(__oid, pda, &__layout, pl_map,	\
+				true));						\
+		pl_obj_layout_free(__layout);					\
+	} while (0)
+
+#define assert_placement_success(pl_map, cid, pda)				\
+	do {									\
+		daos_obj_id_t __oid;						\
+		struct pl_obj_layout *__layout = NULL;				\
+		gen_oid(&__oid, 1, UINT64_MAX, cid);				\
+		assert_success(plt_obj_place(__oid, pda, &__layout, pl_map,	\
+				false));					\
+		pl_obj_layout_free(__layout);					\
+	} while (0)
+
+#define assert_invalid_param(pl_map, cid, pda)					\
+	do {									\
+		daos_obj_id_t __oid;						\
+		struct pl_obj_layout *__layout = NULL;				\
+		int rc;								\
+		gen_oid(&__oid, 1, UINT64_MAX, cid);				\
+		rc = plt_obj_place(__oid, pda,&__layout,			\
```

Comment:
(style)  space required after that ',' (ctx:VxO)
(style)  space required before that '&' (ctx:OxV)